src/hotspot/share/runtime/thread.inline.hpp

```diff
@@ -124,7 +124,9 @@ inline void JavaThread::clear_obj_deopt_flag() {
 
 inline bool JavaThread::clear_async_exception_condition() {
   bool ret = has_async_exception_condition();
-  clear_suspend_flag(_has_async_exception);
```

Comment:
I gotta wonder why the baseline code always called `clear_suspend_flag()`
and did that have any bad side effects?

Reply:
Clearing a cleared bit has no affect and in baseline we are always going to deliver the pending_async_exception, so if it is set then it must be cleared. Now, if delivery is deferred, we must not clear the bit else we lose the fact we may still have to deliver at some point.